### PR TITLE
Update rebase to use ChangesetReader/ChangesetWriter

### PR DESCRIPTION
**Pull request by wonder-sk**
Getting rid of sqlite3session code...

### CODE REVIEW

File: geodiff/src/geodiffrebase.cpp [hunk 1]
```diff
@@ -321,326 +284,230 @@ int _find_mapping_for_new_changeset( const Buffer &buf,
...
-  const RebaseMapping &mapping,
-  unsigned char *aiFlg,
-  std::shared_ptr<BinaryStream> out
-)
+bool _handle_insert( const ChangesetEntry &entry, const RebaseMapping &mapping, ChangesetEntry &outEntry )
 {
-  // first write operation type (iType)
-  out->put( SQLITE_INSERT );
-  out->put( 0 );
+  size_t numColumns = entry.table->primaryKeys.size();
```
Comment (PeterPetrik) [inline geodiff/src/geodiffrebase.cpp:289]: number of columms is number of primary keys?

File: geodiff/src/geodiffutils.hpp [hunk 2]
```diff
@@ -226,7 +227,8 @@ bool has_same_table_schema( std::shared_ptr<Sqlite3Db> db,
                             const std::string &tableName,
                             std::string &errStr );
 
-void get_primary_key( Sqlite3ChangesetIter &pp, int pOp, int &fid, int &nColumn );
+struct ChangesetEntry;
```
Comment (PeterPetrik) [inline geodiff/src/geodiffutils.hpp:230]: move forward decl to the beginning of the file?

File: geodiff/src/geodiffrebase.cpp (hunk 1 above)
Comment (wonder-sk) [reply geodiff/src/geodiffrebase.cpp:289]: primaryKeys array is an array of bools - each column is true/false depending on whether it is part of pkey or not. So the size of the array == number of columns.

Comment (PeterPetrik) [reply geodiff/src/geodiffrebase.cpp:289]: would it be possible to create new function in entry.table to return number of columns. would be much more obvious

**Pull request merged**